src/components/NanoContract/NanoContractDetails.js

```diff
@@ -86,6 +86,8 @@ export const NanoContractDetails = ({ nc }) => {
        * it will be set during the load.
        */
       dispatch(nanoContractHistoryRequest({ ncId: nc.ncId, after: null }));
+    } else {
+      dispatch(nanoContractHistoryRequest({ ncId: nc.ncId, after }))
```

Comment:
You should add `after` as a dependency for the `useEffect`

Reply:
If we do this, the `nanoContractHistoryRequest` will be dispatched twice. The other side effect would be the dispatch of `nanoContractHistoryRequest` until we react the newest tx, however it would happen without a user action, which is not good for usability. `txHistory` already provokes a new rendering if it change, I think.

Reply:
I have refactored it, can you take a new look?